packages/contracts-bedrock/src/universal/StorageSetter.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { ISemver } from "src/universal/ISemver.sol";
+import { Storage } from "src/libraries/Storage.sol";
+
+/// @title StorageSetter
+/// @notice A simple contract that allows setting arbitrary storage slots.
+///         WARNING: this contract is not safe to be called by untrusted parties.
+///         It is only meant as an intermediate step during upgrades.
+contract StorageSetter is ISemver {
```

Comment:
I'm having trouble understanding the flow in which this contract will be used. It has a bunch of public setter functions, but what calls them?
Will it be inherited by a contract which is customized for the specific contract being upgraded?

Reply:
Right now a single upgrade and call is hand coded, keeping it simple because that is all we really need for now, see the logic in https://github.com/ethereum-optimism/optimism/pull/7612